bracket-matcher/bracket_matcher.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+"""Bracket_Matcher.ipynb
+
+Automatically generated by Colab.
+
+Original file is located at
+    https://colab.research.google.com/drive/1xkmSrQubXSBOP286JE_vvdYFQC_WT7E3
+
+#Exercise 3
+"""
+
+#Traverse the text string in search of
+#left and right paranthesis
+#if a left paranthesis is found we traverse
+#the back of the text to find a right paranthesis
+
+
+def BracketMatcher(text):
```

Comment:
Extra credit since this seems to be a more complete implementation than most (including whoever came up with the solution) came up with, which is just to count the number of open and close parentheses. This will handle the case of starting with a closed parenthesis that can never be matched. Nice!